app/controllers/admin/companies_controller.rb

```diff
@@ -7,4 +7,11 @@ def destroy
     Company.find(params[:id]).destroy
     redirect_to admin_dashboard_index_path
   end
+
+  def update
+    company = Company.find(params[:id])
+    company.update(status: params[:company][:status])
+    company.save
+    redirect_to admin_company_path(company.id)
+  end
 end
```

Comment:
I think this works! I like it much better without conditionals it looks very clean!